hal/src/sercom/v2/spi/pads_thumbv7em.rs

```diff
@@ -0,0 +1,385 @@
+//! Define a container for a set of SERCOM pads
+//!
+//! See the [spi module](super) documentation for more details on declaring and
+//! instantiating a [`Pads`] type.
+
+use core::marker::PhantomData;
+
+use crate::gpio::v2::AnyPin;
+use crate::sercom::v2::*;
+use crate::typelevel::{NoneT, Sealed};
+
+use super::{Capability, Duplex, Rx, Tx};
+
+//=============================================================================
+// Dipo
+//=============================================================================
+
+/// Map an [`OptionalPadNum`] to its corresponding `DIPO` value
+pub trait Dipo: OptionalPadNum {
+    const DIPO: Option<u8>;
+}
+
+impl Dipo for NoneT {
+    const DIPO: Option<u8> = None;
+}
+
+impl Dipo for Pad0 {
+    const DIPO: Option<u8> = Some(0);
+}
+impl Dipo for Pad1 {
+    const DIPO: Option<u8> = Some(1);
+}
+impl Dipo for Pad2 {
+    const DIPO: Option<u8> = Some(2);
+}
+impl Dipo for Pad3 {
+    const DIPO: Option<u8> = Some(3);
+}
```

Comment:
This is new. Is there a downside to just setting `DIPO/DOPO` to a nonsensical value instead? (asking for `uart-v2`).